src/platform/packages/private/kbn-grid-layout/grid/grid_row/grid_row_header.test.tsx

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+import { cloneDeep } from 'lodash';
+import React from 'react';
+
+import { EuiThemeProvider } from '@elastic/eui';
+import { RenderResult, act, render, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+import { gridLayoutStateManagerMock, mockRenderPanelContents } from '../test_utils/mocks';
+import { getSampleLayout } from '../test_utils/sample_layout';
+import { GridLayoutStateManager } from '../types';
+import { GridRowHeader, GridRowHeaderProps } from './grid_row_header';
+import { GridLayoutContext, GridLayoutContextType } from '../use_grid_layout_context';
+
+const toggleIsCollapsed = jest
+  .fn()
+  .mockImplementation((rowIndex: number, gridLayoutStateManager: GridLayoutStateManager) => {
+    const newLayout = cloneDeep(gridLayoutStateManager.gridLayout$.value);
+    newLayout[rowIndex].isCollapsed = !newLayout[rowIndex].isCollapsed;
+    gridLayoutStateManager.gridLayout$.next(newLayout);
+  });
+
+describe('GridRowHeader', () => {
+  const renderGridRowHeader = (
+    propsOverrides: Partial<GridRowHeaderProps> = {},
+    contextOverrides: Partial<GridLayoutContextType> = {}
+  ) => {
+    return render(
+      <EuiThemeProvider>
+        <GridLayoutContext.Provider
+          value={
+            {
+              renderPanelContents: mockRenderPanelContents,
+              gridLayoutStateManager: gridLayoutStateManagerMock,
+              ...contextOverrides,
+            } as GridLayoutContextType
+          }
+        >
+          <GridRowHeader
+            rowIndex={0}
+            toggleIsCollapsed={() => toggleIsCollapsed(0, gridLayoutStateManagerMock)}
+            collapseButtonRef={React.createRef()}
+            {...propsOverrides}
+          />
+        </GridLayoutContext.Provider>
+      </EuiThemeProvider>
+    );
+  };
+
+  beforeEach(() => {
+    toggleIsCollapsed.mockClear();
+    act(() => {
```

Comment:
I didn't realize that RTL had an `act` function. What is the reasoning behind using this? Is it so that the `next` event triggers a re-render?

Reply:
Correct, yup! `next` is essentially acting as a user action here, and so I wanted to [wait for the DOM to update](src/platform/packages/private/kbn-grid-layout/grid/grid_row/grid_row_header.test.tsx) before I move on to the next test.